src/tiles/storage.rs

```diff
@@ -1,26 +1,37 @@
 use bevy::prelude::*;
 
 use crate::map::TilemapSize;
+use crate::TilemapMeshType;
 
 use super::TilePos;
 
 /// Used to store tile entities for fast look up.
 /// Tile entities are stored in a grid. The grid is always filled with None.
 #[derive(Component, Default, Debug, Clone)]
-pub struct TileStorage {
+pub struct TilemapStorage {
+    mesh_type: TilemapMeshType,
```

Comment:
Storing `TilemapMeshType` here means that we hard lock on requiring a `TilemapStorage`. Although `TilemapStorage` being optional is not very well documented.